drivers/accel/adxl367/adxl367.c

```diff
@@ -0,0 +1,1310 @@
+{
+	int32_t ret;
+
+	if (dev->comm_type == ADXL367_SPI_COMM)
+		ret = no_os_spi_remove(dev->spi_desc);
+	else
+		ret = no_os_i2c_remove(dev->i2c_desc);
+
+	free(dev);
+
+	return ret;
+}
+
+/***************************************************************************//**
+ * @brief Performs self test.
+ *
+ * @param dev - The device structure.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t adxl367_self_test(struct adxl367_dev *dev)
+{
+	int32_t ret;
+	int16_t x_axis_1, x_axis_2, dif, min, max;
+	uint8_t read_val;
+
+	ret = adxl367_set_power_mode(dev, ADXL367_OP_MEASURE);
+	if (ret)
+		return ret;
+
+	ret = adxl367_reg_write_msk(dev, ADXL367_REG_SELF_TEST, ADXL367_SELF_TEST_ST,
+				    ADXL367_SELF_TEST_ST);
+	if (ret)
+		return ret;
+	// 4 / lowest ODR = 80ms
+	no_os_mdelay(80);
+	ret = adxl367_get_register_value(dev, &read_val, ADXL367_REG_XDATA_H, 1);
+	if (ret)
+		return ret;
+	x_axis_1 = read_val << 6;
+	ret = adxl367_get_register_value(dev, &read_val, ADXL367_REG_XDATA_L, 1);
+	if (ret)
+		return ret;
+	x_axis_1 += read_val >> 2;
+	// extend sign to 16 bits
+	if (x_axis_1 & NO_OS_BIT(13))
+		x_axis_1 |= NO_OS_GENMASK(15, 14);
+
+	ret = adxl367_reg_write_msk(dev, ADXL367_REG_SELF_TEST,
+				    ADXL367_SELF_TEST_ST_FORCE, ADXL367_SELF_TEST_ST_FORCE);
+	if (ret)
+		return ret;
+	// 4 / lowest ODR = 80ms
+	no_os_mdelay(80);
+	ret = adxl367_get_register_value(dev, &read_val, ADXL367_REG_XDATA_H, 1);
+	if (ret)
+		return ret;
+	x_axis_2 = read_val << 6;
+	ret = adxl367_get_register_value(dev, &read_val, ADXL367_REG_XDATA_L, 1);
+	if (ret)
+		return ret;
+	x_axis_2 += read_val >> 2;
+	// extend sign to 16 bits
+	if (x_axis_1 & NO_OS_BIT(13))
+		x_axis_1 |= NO_OS_GENMASK(15, 14);
+
+	ret = adxl367_set_power_mode(dev, ADXL367_OP_STANDBY);
+	if (ret)
+		return ret;
+
+	ret = adxl367_reg_write_msk(dev, ADXL367_REG_SELF_TEST, 0,
+				    ADXL367_SELF_TEST_ST_FORCE | ADXL367_SELF_TEST_ST);
+	if (ret)
+		return ret;
+
+	dif = x_axis_2 - x_axis_1;
+	min = 90 * 100 / 25 * adxl367_scale_mul[dev->range];
+	max = 270 * 100 / 25 * adxl367_scale_mul[dev->range];
+
+	if ((dif >= min) && (dif <= max)) {
+		pr_info("ADXL367 passed self-test\n");
+		return 0;
+	} else {
+		pr_err("%s: Failed self-test.\n", __func__);
+		return -1;
+	}
+}
+
+/***************************************************************************//**
+ * @brief Writes data into a register.
+ *
+ * @param dev              - The device structure.
+ * @param register_value   - Data value to write.
+ * @param register_address - Address of the register.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t adxl367_set_register_value(struct adxl367_dev *dev,
+				   uint8_t register_value,
+				   uint8_t register_address)
+{
+	uint8_t buffer[3] = {0, 0, 0};
```

Comment:
You could just do `uint8_t buffer[3] = {0}` and the whole array would be initialized to 0. This way, you also avoid using memset.